### PR TITLE
removed dependency on SandboxConfig from ApiServices

### PR DESCRIPTION
**Pull request by gaboraranyossy-da**
See title.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/main/scala/com/digitalasset/ledger/server/apiserver/ApiServices.scala [hunk 1]
```diff
@@ -12,21 +12,21 @@ import com.daml.ledger.participant.state.index.v2.{
   _
 }
 import com.daml.ledger.participant.state.v2.WriteService
+import com.daml.ledger.participant.state.v1.TimeModel
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/ledger/server/apiserver/ApiServices.scala]: Use `v2` of the participant api. Ideally also remove the dependency on v1 altogether, like we did for the index-api v2.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala [hunk 2]
```diff
@@ -162,10 +162,14 @@ class SandboxServer(actorSystemName: String, config: => SandboxConfig) extends A
...
         case (None, _) =>
-          val ts = TimeServiceBackend.simple(Instant.EPOCH)
+          val ts = TimeServiceBackend.simple(
+            Instant.EPOCH,
+            config.timeProviderType == TimeProviderType.StaticAllowBackwards)
           (ts, Some(ts))
         case (Some(ledgerTime), _) =>
-          val ts = TimeServiceBackend.simple(ledgerTime)
+          val ts = TimeServiceBackend.simple(
+            ledgerTime,
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala]: You can combine the last 2 branches and simply do `mbLedgerTime.getOrElse(Instant.EPOCH)`.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/ApiSubmissionService.scala [hunk 3]
```diff
@@ -12,6 +12,7 @@ import com.daml.ledger.participant.state.v2.{
   TransactionMeta,
   WriteService
 }
+import com.daml.ledger.participant.state.v1._
```
Comment (gerolf-da) [inline ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/ApiSubmissionService.scala]: Use `participant.state.v2._`

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/services/command/time/TimeModelValidator.scala [hunk 4]
```diff
@@ -5,9 +5,9 @@ package com.digitalasset.platform.server.services.command.time
 
 import java.time.{Duration, Instant}
 
-import com.daml.ledger.participant.state.v1.TimeModelChecker
+import com.daml.ledger.participant.state.v1.TimeModel
```
Comment (gerolf-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/services/command/time/TimeModelValidator.scala]: Use `participant.state.v2`.

**Comment by gaboraranyossy-da** [comment]
@gerolf-da some of these TimeModel related things are simply grabbed within a few classes like [InMemoryKVParticipantState](https://github.com/digital-asset/daml/blob/master/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InMemoryKVParticipantState.scala#L89), [KeyValueCommitting](https://github.com/digital-asset/daml/blob/master/ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala#L17 ) etc.. In those packages I guess we should still use `v1`. I'd rather avoid mapping between these identical types.

**Comment by gerolf-da** [comment]
Yes, those classes need to be copied to a backport folder. @mziolekda, can you coordinate with @gaboraranyossy-da on this effort please, as you are very familiar with the kvutils code-base.

File: ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala (hunk 2 above)
Comment (gaboraranyossy-da) [reply ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala]: right, will do that.

**Pull request merged**